src/geo/locate.js

```diff
@@ -40,6 +44,68 @@ const handleQuery = async (opts) => {
   }
 }
 
+const locateCity = async ({ city, region, country }) => {
+  const query = {
+    text: `${city}, ${region} ${country}`,
+    size: 1
+  }
+
+  // check if cache has it first
+  const lruKey = JSON.stringify(query)
+  if (lru.has(lruKey)) return lru.get(lruKey)
+
+  const opts = {
+    query,
+    host: pelias.hosts.search
+  }
+  // not in cache, fetch it
+  const out = handleQuery(opts)
+  if (!out) return
+  // put it in cache for later
+  lru.set(lruKey, out)
+  return out
+}
+
+const runOverpassQuery = async (query) => {
+  const qs = `[out:json];
+    ${query}
+  out body;`
+  const { body } = await request
+    .post('http://overpass-api.de/api/interpreter')
+    .send(qs)
+    .retry(10)
+    .agent(agent)
+  return body
+}
+
+const locateWay = async ({ street, bbox }) => {
+  const query = `way
+      ["name"~"${capitalize.words(street)}"]
+      (${bbox[1]}, ${bbox[0]}, ${bbox[3]}, ${bbox[2]});`
+  return runOverpassQuery(query)
+}
+
+const lookupNodeId = async (nodeId) => {
+  const { elements } = await runOverpassQuery(`node(${nodeId});`)
```

Comment:
can this be undefined?

Reply:
Since it's only called with node id's retrieved from the overpass API, no. However, it _could_ technically be undefined.